### PR TITLE
kasmweb: 1.12.0 -> 1.15.0

### PR DESCRIPTION
**Pull request by s1341**
Bump kasmweb fro 1.12.0 to 1.15.0

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by LudovicoPiero** [comment]
Result of `nixpkgs-review pr 246777` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>kasmweb</li>
  </ul>
</details>

File: nixos/modules/services/web-apps/kasmweb/default.nix [hunk 1]
```diff
@@ -124,29 +124,20 @@ in
...
-        ];
+        wants = ["network-online.target"];
+        after = ["network-online.target"];
         serviceConfig = {
           Type = "oneshot";
           ExecStart = pkgs.substituteAll {
             src = ./initialize_kasmweb.sh;
             isExecutable = true;
-            binPath = lib.makeBinPath [ pkgs.docker pkgs.openssl pkgs.gnused ];
+            binPath = lib.makeBinPath [ pkgs.docker pkgs.openssl pkgs.gnused pkgs.yq-go ];
```
Comment (h3ndrik) [inline nixos/modules/services/web-apps/kasmweb/default.nix:140]: Sorry, I'm not an expert. Does this specifically depend on docker? is there a way to depend on either docker or the podman compatibility docker command?

**Review by h3ndrik — COMMENTED** [review]
Hey, I hope this gets merged soon. I have an additional comment / question:

Is it possible to run this with podman's docker compatibility mode?

```nix
  virtualisation.podman = {
    enable = true;
    dockerCompat = true;
    dockerSocket.enable = true;
    defaultNetwork.settings.dns_enabled = true;
  };
  virtualisation.oci-containers.backend = lib.mkForce "podman";
```

**Comment by s1341** [comment]
IIRC I tried to switch to podman but there were issues. Possibly with the network setup?

**Comment by cig0** [comment]
Hi all, is this PR still viable?

**Comment by s1341** [comment]
Yes. SFAIK.

**Review by SuperSandro2000 — COMMENTED** [review]
Please clean up the commit history and drop the merge commit.

File: nixos/modules/services/web-apps/kasmweb/initialize_kasmweb.sh [hunk 2]
```diff
@@ -118,9 +113,9 @@ sed -i -e "s/GUACTOKEN/@defaultGuacToken@/g" \
     -e "s/APIHOSTNAME/proxy/g" \
     @datastorePath@/conf/app/kasmguac.app.config.yaml
 
-sed -i -e "s/GUACTOKEN/@defaultGuacToken@/g" \
-    -e "s/APIHOSTNAME/proxy/g" \
-    @datastorePath@/conf/database/seed_data/default_connection_proxies.yaml
+# sed -i -e "s/GUACTOKEN/@defaultGuacToken@/g" \
```
[the remark this answers is not in the document]
Comment (s1341) [reply nixos/modules/services/web-apps/kasmweb/initialize_kasmweb.sh]: woops. Resolved.

File: nixos/modules/services/web-apps/kasmweb/initialize_kasmweb.sh [hunk 3]
```diff
@@ -60,11 +55,11 @@ else
     echo "export SERVER_ID=$SERVER_ID" >> @datastorePath@/ids.env
 
     mkdir -p @datastorePath@/certs
-    openssl req -x509 -nodes -days 1825 -newkey rsa:2048 -keyout @datastorePath@/certs/kasm_nginx.key -out @datastorePath@/certs/kasm_nginx.crt -subj "/C=US/ST=VA/L=None/O=None/OU=DoFu/CN=$(hostname)/emailAddress=none@none.none" 2> /dev/null
+    openssl req -x509 -nodes -days 1825 -newkey rsa:2048 -keyout @datastorePath@/certs/kasm_nginx.key -out @datastorePath@/certs/kasm_nginx.crt -subj "/C=US/ST=VA/L=None/O=None/OU=DoFu/CN=$(@coreutils@/bin/hostname)/emailAddress=none@none.none" 2> /dev/null
```
[the remark this answers is not in the document]
Comment (s1341) [reply nixos/modules/services/web-apps/kasmweb/initialize_kasmweb.sh]: ok. resolved.

File: nixos/modules/services/web-apps/kasmweb/default.nix
[the remark this answers is not in the document]
Comment (s1341) [reply nixos/modules/services/web-apps/kasmweb/default.nix:1]: I cleaned up the commit history.

**Comment by s1341** [comment]
@SuperSandro2000 Care to re-review?

**Comment by s1341** [comment]
thanks!

**Pull request merged**